tests/test_lab_data_list.py

```diff
@@ -0,0 +1,110 @@
+    test_dir = tmp_path / "list_datasets_test_dir"
+    test_dir.mkdir()
+
+    files_to_create = [
+        "test_modelA_2024-10-27T08_34_48.jsonl",
+        "train_modelA_2024-10-27T08_34_48.jsonl",
+        "messages_modelB_2024-10-26T21_32_32.jsonl",
+        "skills_train_msgs_2024-10-27T08_34_48.jsonl",
+    ]
+
+    file_info = []
+    for filename in files_to_create:
+        file_path = test_dir / filename
+        file_path.write_text("test content")
+        stat = file_path.stat()
+        created_at = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime(stat.st_ctime))
+        formatted_size = f"{12.00:.2f} B"
+        file_info.append((filename, created_at, formatted_size))
+
+    node_datasets_dir = test_dir / "node_datasets_2024-10-27T08_34_48"
+    node_datasets_dir.mkdir()
+    node_file = node_datasets_dir / "compositional_skills_valid.jsonl"
+    node_file.write_text("test content")
+    node_stat = node_file.stat()
+    node_created_at = time.strftime(
+        "%Y-%m-%d %H:%M:%S", time.localtime(node_stat.st_ctime)
+    )
+    node_formatted_size = f"{12.00:.2f} B"
+
+    result = cli_runner.invoke(
+        lab.ilab,
+        [
+            "--config=DEFAULT",
+            "data",
+            "list",
+            "--dataset-dirs",
+            str(test_dir),
+        ],
+    )
+
+    expected_output = textwrap.dedent(f"""
+        modelA 2024-10-27 08:34:48
+        +--------------------------------------------------------------------+---------------------+-----------+
+        | Dataset                                                            | Created At          | File size |
+        +--------------------------------------------------------------------+---------------------+-----------+
+        | test_modelA_2024-10-27T08_34_48.jsonl                              | {file_info[0][1]} | {file_info[0][2]} |
+        | train_modelA_2024-10-27T08_34_48.jsonl                             | {file_info[1][1]} | {file_info[1][2]} |
+        | skills_train_msgs_2024-10-27T08_34_48.jsonl                        | {file_info[3][1]} | {file_info[3][2]} |
+        | node_datasets_2024-10-27T08_34_48/compositional_skills_valid.jsonl | {node_created_at} | {node_formatted_size} |
+        +--------------------------------------------------------------------+---------------------+-----------+
+
+        modelB 2024-10-26 21:32:32
+        +-------------------------------------------+---------------------+-----------+
+        | Dataset                                   | Created At          | File size |
+        +-------------------------------------------+---------------------+-----------+
+        | messages_modelB_2024-10-26T21_32_32.jsonl | {file_info[2][1]} | {file_info[2][2]} |
+        +-------------------------------------------+---------------------+-----------+
+    """).strip()
+
+    assert result.exit_code == 0
+    assert sorted(extract_data_entries(result.output)) == sorted(
```

Comment:
Why is the sorting needed here?

Reply:
Seems different order from different os, one of failed tests from ubuntu. so sort it.

Reply:
I might be reading it wrong, but I think this might be because you are sorting models but not the datasets?  If that's correct, was there a reason not to sort both?

Reply:
thanks, updated